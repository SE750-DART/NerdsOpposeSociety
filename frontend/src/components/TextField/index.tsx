import React from "react";
import styles from "./style.module.css";

interface TextFieldProps {
  label: string;
  textValue: string;
  size?: "big" | "small";
  onChangeHandler: (arg0: string) => void;
  disabled?: boolean;
}

const TextField = ({
  label,
  textValue,
  size = "big",
  onChangeHandler,
  disabled = false,
}: TextFieldProps) => {
  let sizeStyle: string = "";
  if (size === "big") {
    sizeStyle = styles.textfield__big;
  } else if (size === "small") {
    sizeStyle = styles.textfield__small;
  }

  return (
    <input
      className={`${styles.textfield} ${sizeStyle}`}
      placeholder={label}
      value={textValue}
      type="text"
      onChange={(e) => onChangeHandler(e.target.value)}
      disabled={disabled}
    />
  );
};

export default TextField;
