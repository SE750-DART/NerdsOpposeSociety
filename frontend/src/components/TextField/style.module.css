.textfield {
  align-items: center;
  border: none;
  border-radius: 16px;
  box-shadow: inset 0px 0px 10px rgba(0, 0, 0, 0.25);
  display: flex;
  font-family: "DM Sans", sans-serif;
  font-size: 32px;
  font-weight: bold;
  justify-content: center;
  padding: 0 24px;
  text-align: center;
  transition: 0.3s;

  height: 64px;
  width: 100%;
}
