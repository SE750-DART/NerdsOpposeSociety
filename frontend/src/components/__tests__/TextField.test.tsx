import React from "react";
import renderer from "react-test-renderer";
import TextField from "../TextField";

it("Text field snapshot testing", () => {
  const treeBig = renderer
    .create(
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      <TextField textValue="" label="Game code" onChangeHandler={() => {}} />
    )
    .toJSON();
  expect(treeBig).toMatchSnapshot();

  const treeSmall = renderer
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    .create(<TextField textValue="" onChangeHandler={() => {}} size="small" />)
    .toJSON();
  expect(treeSmall).toMatchSnapshot();
});
