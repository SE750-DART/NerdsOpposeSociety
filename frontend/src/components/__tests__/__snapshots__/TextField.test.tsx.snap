// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text field snapshot testing 1`] = `
<input
  className="inputField inputField__big"
  disabled={false}
  onChange={[Function]}
  placeholder="Game code"
  type="text"
  value=""
/>
`;

exports[`Text field snapshot testing 2`] = `
<input
  className="inputField inputField__small"
  disabled={false}
  onChange={[Function]}
  placeholder=""
  type="text"
  value=""
/>
`;
